### PR TITLE
feat: static modules

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Convert generated to static modules

## Context:
#8647 #6747

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by JamieMagee** [comment]
Is generate-imports required anymore?

**Comment by viceice** [comment]
Yes, for node schedule and pip_setup extractor. Maybe it should be renamed. 😏

File: tools/generate-imports.ts [hunk 1]
```diff
@@ -108,24 +108,24 @@ async function generateData(): Promise<void> {
...
+    //   types: ['Platform'],
+    //   excludes: ['utils', 'git'],
+    // });
+
+    // // versioning
+    // await generate({
+    //   path: 'versioning',
+    //   types: ['VersioningApi', 'VersioningApiConstructor'],
+    //   map: '.api',
+    // });
```
Comment (rarkins) [inline tools/generate-imports.ts:128]: Delete thesee

**Review by rarkins — CHANGES_REQUESTED** [review]
We should add a test for each of these modules to make sure that all (relevant) subdirectories are imported and match the import name.

File: tools/generate-imports.ts (hunk 1 above)
Comment (viceice) [reply tools/generate-imports.ts:128]: sure

**Comment by viceice** [comment]
Strange, not sure why it fails with oom. 🤔

**Review by JamieMagee — COMMENTED** [review]
Ugh, leaving a comment from the GitHub app starts a full review.

EDIT: I left this comment earlier but had to 'approve' it on web. I have no idea what's going on here.

File: tools/generate-imports.ts [hunk 2]
```diff
@@ -108,24 +108,24 @@ async function generateData(): Promise<void> {
     await generateData();
 
     // datasources
-    await generate({ path: 'datasource', types: ['DatasourceApi'] });
+    // await generate({ path: 'datasource', types: ['DatasourceApi'] });
```
Comment (JamieMagee) [inline tools/generate-imports.ts:111]:
```suggestion
```

File: tools/generate-imports.ts (hunk 1 above)
Comment (JamieMagee) [inline tools/generate-imports.ts:128]:
```suggestion
```

**Comment by rarkins** [comment]
Yes, happens to me often. It defaults to pending comments in a review, not sure if you can change that

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs 2 missing managers added :)

**Comment by viceice** [comment]
> We should add a test for each of these modules to make sure that all (relevant) subdirectories are imported and match the import name.

https://github.com/renovatebot/renovate/blob/21ebb743493ebb37b82749a282e70c573cc54502/lib/datasource/index.spec.ts#L49
https://github.com/renovatebot/renovate/blob/53ca91301f6a36c0c87e7d6b06b1bd8c9ec1b283/lib/manager/index.spec.ts#L35
https://github.com/renovatebot/renovate/blob/04d2cd1514d8f7e84844e2c5e6091478f59c21db/lib/platform/index.spec.ts#L29
https://github.com/renovatebot/renovate/blob/53ce3cf4d6bc2995aa2f95fdeb473873f09f5e69/lib/versioning/index.spec.ts#L38

Already tested 🙃

**Review by rarkins — APPROVED** [review]

**Pull request merged**